gax/src/main/java/com/google/api/gax/rpc/Callables.java

```diff
@@ -226,10 +241,26 @@ private BatchingCreateResult(
         initialCallable, scheduler, longRunningClient, operationCallSettings);
   }
 
+  /**
+   * areRetrtiesDisabled determines if retries are disabled for an RPC based on the set of retryable
+   * codes, the retry settings, and the overall timeout.
+   *
+   * @param retryableCodes the set of {@link StatusCode.Code}s configured to be retried. If empty,
+   *     retries are disabled.
+   * @param retrySettings the {@link RetrySettings} for the RPC. When maxAttempts is 1, or
+   *     maxAttempts and totalTimeout are 0 and overallTimeout is unset, retries are disabled.
+   * @param overallTimeout the default overall timeout configured for the call. When it is null, and
+   *     maxAttempts and totalTimeout are both 0, retries are disabled.
+   * @return {@code true} if retries are enabled and {@code false} if not.
+   */
   private static boolean areRetriesDisabled(
-      Collection<StatusCode.Code> retryableCodes, RetrySettings retrySettings) {
+      Collection<StatusCode.Code> retryableCodes,
+      RetrySettings retrySettings,
+      Duration overallTimeout) {
     return retrySettings.getMaxAttempts() == 1
         || retryableCodes.isEmpty()
-        || (retrySettings.getMaxAttempts() == 0 && retrySettings.getTotalTimeout().isZero());
+        || (retrySettings.getMaxAttempts() == 0
+            && retrySettings.getTotalTimeout().isZero()
+            && overallTimeout == null);
```

Comment:
I'm not a fan of the RetrySettings maxAttempts and totalTimeout determining if an RPC is retryable or not. If these values are `0` (the unset default) we should let the `shouldRetry` logic of the TimedRetryAlgorithm decide what to do. But this predates me, and I added a check for overallTimeout being unset in order to keep backwards compat when we stop generating RetrySettings.totalTimeout in the gapic by default (which means both maxAttempts and totalTimeout will both be 0 in a newly generated gapic).